packages/medusa/src/api/routes/admin/discounts/__tests__/create-discount.js

```diff
@@ -2,7 +2,15 @@ import { IdMap } from "medusa-test-utils"
 import { request } from "../../../../../helpers/test-request"
 import { DiscountServiceMock } from "../../../../../services/__mocks__/discount"
 
+const validRegionId = IdMap.getId("region-france")
+
 describe("POST /admin/discounts", () => {
+  const generateAdminJwtSession = () => ({
```

Comment:
suggestion: this can be a simple object instead of a function. something like `const adminSession = {}` and then in the header you can have 
```ts
{ 
  // ...
  adminSession 
}
```

Reply:
I started it as a variable, but the object was being mutated on every request, making it fail for other tests. I wanted to bring this up since this might cause other unexpected side-effects. 

Here is what it looks like. 

1st test - ```generateAdminJwtSession -  { jwt: { userId: 'true_kWRj8pshjp' } }```

2nd test - ```
generateAdminJwtSession -  {
      jwt: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiJ0cnVlX2tXUmo4cHNoanAiLCJpYXQiOjE2NzAzNjAyODYsImV4cCI6MTY3MDM2MjA4Nn0.7Upf0C_YZZuJH9ERcSx9NCAiW3N2TffB3caHUsbkQ9c'
    }
```

Reply:
Regarding the other concern - `We should probably update the service method that creates the discounts to enforce at least one` - I've added a test case for this in the new commit.

It's already guarding against that case - let me know if theres a better way to handle it.

Reply:
If you store in a const  this

         const adminSession = {
            jwt: {
              userId: IdMap.getId("admin_user"),
            },
          }, 
          
above the first describe the value change between the calls? I might be wrong.
Since it is non blocking let it like that 👍

Reply:
> Regarding the other concern - `We should probably update the service method that creates the discounts to enforce at least one` - I've added a test case for this in the new commit.
> 
> It's already guarding against that case - let me know if theres a better way to handle it.

Regarding the test cases you have added, it is very nice from an api point of view. It ensure that the api requires the regions. 

But from the service point of view, we can still create a discount without any region and, this is what I am pointing out. It means that a consumer using the service could be able to create a discount without any region by calling the service. Am I right? In that case, we can still have an unstable state for a discount. Let me know if i am missing something.

Reply:
> If you store in a const this
> 
> ```
>      const adminSession = {
>         jwt: {
>           userId: IdMap.getId("admin_user"),
>         },
>       }, 
> ```
> 
> above the first describe the value change between the calls? I might be wrong. Since it is non blocking let it like that 👍

Yup, even after storing in a const

Reply:
> But from the service point of view, we can still create a discount without any region and, this is what I am pointing out. It means that a consumer using the service could be able to create a discount without any region by calling the service. Am I right? In that case, we can still have an unstable state for a discount. Let me know if i am missing something.

Ah I see what you mean now, good catch. I'll add that in.